zulip/zulip
# notifications: Rename `private_message` to `direct_message`.

[CZO message](https://chat.zulip.org/#narrow/stream/2-general/topic/private.20messages.20.3D.3E.20direct.20messages/near/1588736):
> It's probably be a while before we manage to rename everything, but I think we should do any renames that are easy to verify as we notice them. (Some of the backend stuff is tricky, like the notification triggers, because items can be in the API, database, or queues -- **@Prakhar-Pratyush** FYI for the `private_message` notification trigger value that sits in queues, mainly since you're doing a similar rename right now for topic wildcard mentions and it might make sense for you to change them both while it's fresh)

---
* The First commit renames the `PRIVATE_MESSAGE` attribute of the `NotificationTriggers` class to `DIRECT_MESSAGE`. -- custom migration to update the existing value in the database + API changes.
* The Second commit renames the keyword `pm` to `dm` in the `RecipientInfoResult` dataclass. -- handles the fields that sit in the queues.



---
**Manual Testing**

I did manual testing by adding the translation code (`TODO/compatibility`) in `event_queue.py` and keeping the field in the event sent by `message_send` and `message_edit` codepath as `pm_mention_email_disabled_user_ids` (instead of `dm_mention_email_disabled_user_ids`).

* Disabled the setting "email notification for DM, mentions, and alerts"

* No missed message emails were sent.

* Without the compatibility code i.e. simply using the following code block,
``` 
dm_mention_push_disabled_user_ids = set(event_template.pop("dm_mention_push_disabled_user_ids", []) 
```
missed message emails were sent.

---

<details>
<summary>Self-review checklist</summary>





      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.


Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).


Completed manual review and testing of the following:

</details>
